### DISCUSSION TITLE
Finding Plateaus in TS data

### DISCUSSION THREAD

**Discussion by Mopele** [TDAmeritrade/stumpy#370] [Q&A]
Hey, I want to find plateaus in my time series data. As I understand stumpy, it is a tool to find patterns and anomalies. But can I tell it somehow to find plateaus? Is there a Example for that?

Thanks!

**Comment by seanlaw** [comment]
> Hey, I want to find plateaus in my time series data. As I understand stumpy, it is a tool to find patterns and anomalies. But can I tell it somehow to find plateaus? Is there a Example for that?
> 
> 
> 
> Thanks!

@Mopele Thank you for the question. Since the term plateau may be somewhat overloaded and can be interpreted in multiple ways, would you mind providing a visual example of what you mean by a "plateau"?

**Reply by Mopele** [reply]
Yes, I will provide visuals. Currently I am detecting "plateaus" by looking at the derivate of the values and if the derivate doesnt spike I asume it is a plateau. 

I have a easy dataset and a hard one. In the hard one you cant see the plateaus good but they look similar to the one in the easy example.

Easy
![bokeh_plot (1)](https://user-images.githubusercontent.com/42115074/112153253-5ed84180-8be3-11eb-9c8b-abeb07ead377.png)

Hard:
![hard data](https://user-images.githubusercontent.com/42115074/112153329-70214e00-8be3-11eb-96bf-2e9a1337916f.png)

**Reply by seanlaw** [reply]
So, given the easy and hard time series above, if we use these as separate inputs to some analysis, would would be the expected output(s)? Would it be a simple threshold value like `200` for the easy case and `5000` for the hard case? 

My initial thought is that STUMPY would not be much help here but I would need to better understand your goal(s).

**Reply by Mopele** [reply]
So right now I have some code that takes the timeseries as input and outputs a list of tupels with a startindex and endindex of a plateau. 

Would such a result be possible with stumpy? 

It would also help if I could somehow find the amplitude of the signals just for the flat parts. This would also get me better results.

I thought that with stumpy I could give it a flat section and it could output me similar sections so i can find the amplitude of the sections, but i am still new to stumpy...

**Reply by seanlaw** [reply]
I might be overlooking something obvious but, unfortunately, I don't think this is a great use case for STUMPY.

**Reply by Mopele** [reply]
Ok, thanks! So I guess then stumpy will not help me out so much
